Integrations/python/deephaven/learn/__init__.py

```diff
@@ -0,0 +1,190 @@
+
+    Args:
+        inputs  : list of Inputs to validate.
+        outputs : list of Outputs to validate.
+        table   : table to check Input and Output columns against.
+
+    Raises:
+        ValueError : if at least one of the Input columns does not exist in the table.
+        ValueError : if at least one of the Output columns already exists in the table.
+    """
+
+    inputColumns = [input.input.getColNames()[i] for input in inputs for i in range(len(input.input.getColNames()))]
+
+    if not all(inputColumn in list(table.getMeta().getColumn("Name").getDirect()) for inputColumn in inputColumns):
+        raise ValueError("All Input columns must exist in the given table.")
+
+    if not outputs == None:
+        columns = [*set(inputColumns), *[output.output.getColName() for output in outputs]]
+
+        if len(columns) != len(set(columns)):
+            raise ValueError("Cannot use existing column names for Output columns.")
+
+    return
+
+
+@_passThrough
+def _verifyColumn(column, table):
+    """
+    Ensures that the given column is not present in the table, and modifies it if necessary until it is not in the table.
+
+    Args:
+        column : column name that should not exist in the table.
+        table  : table to check column against.
+
+    Returns:
+        the column name, modified if necessary so that it is not present in the table.
+    """
+
+    tableColumns = list(table.getMeta().getColumn("Name").getDirect())
+
+    if column not in tableColumns:
+        return column
+
+    else:
+        i = 0
+        while column in tableColumns:
+            column = column + str(i)
+
+            if column not in tableColumns:
+                return column
+            i += 1
+
+
+@_passThrough
+def eval(table=None, model_func=None, inputs=[], outputs=[], batch_size = None):
+    """
+    Takes relevant data from Deephaven table using inputs, converts that data to the desired Python type, feeds
+    it to model_func, and stores that output in a Deephaven table using outputs.
+
+    Args:
+        table      : the Deephaven table to perform computations on
+        model_func : python function that performs computations on the table.
+        inputs     : list of Deephaven input objects
+        outputs    : list of Deephaven output objects
+        batch_size : number of rows for which model_func is evaluated at once. Note that this must be provided for a live table.
+
+    Returns:
+        the table with added columns containing the results of evaluating model_func.
+
+    Raises:
+        ValueError : if table is live and no batch size was provided
+    """
+
+    _validate(inputs, outputs, table)
+
+    if batch_size == None:
+        if table.isLive():
+            raise ValueError("Batch size cannot be inferred on a live table. Please specify a batch size.")
+        batch_size = table.size()
+
+    globals()["computer"] = _Computer_(table, model_func, [input.input for input in inputs], batch_size)
```

Comment:
the "computer" variable has potential naming conflicts with user code.  You have two possible solutions:
1) Give it a less probable "__computer" variable name... but you may still have conflicts
2) Give it a unique name and delete the value from globals via a finally block.